sdccc/src/main/java/com/draeger/medical/sdccc/tests/util/MdibHistorian.java

```diff
@@ -205,10 +224,29 @@ public HistorianResult episodicReportBasedHistory(final String sequenceId)
         try {
             final var messages =
                     messageStorage.getInboundMessagesByBodyType(Constants.RELEVANT_REPORT_BODIES.toArray(new QName[0]));
+            final BigInteger[] last = new BigInteger[1];
             final var stream = messages.getStream()
                     .map(this::unmarshallReport)
                     .filter(report -> sequenceId.equals(report.getSequenceId()))
                     .filter(mdibVersionPredicate)
+                    .filter(it -> {
```

Comment:
Stream.filter is a stateless intermediate operation that should not produce side-effects. Each element should be processed independently. Perhaps it is necessary to build a list with e.g. Stream.forEachOrdered and use this list as a new source. Although there are certainly more performant solutions.

Reply:
Good Point! The stateful filter works, but is not thread-safe.

I forced the Stream to be sequential to ensure that it is not processed in parallel.